File: flex/buzzer/c/beep.c
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

//Pin 31 on Raspberry Pi corresponds to BCM GPIO 6 and wiringPi pin 22
#define BeepPin 22

void close()
{
	printf("\nTurning off buzzer...\n");
	digitalWrite(BeepPin, LOW);
	exit(0);
}

int main(void)
{
	signal(SIGINT, close);

	if(-1 == wiringPiSetup())
	{
			printf("setup wiringPi failed!");
			return 1;
	}

	//Set GPIO pin
	pinMode(BeepPin, OUTPUT);

	printf("Press CTRL+C to exit.\n");

	//Play a sound until the user closes the app
	while(1)
	{
		digitalWrite(BeepPin, LOW);
		delay(2);
		digitalWrite(BeepPin, HIGH);
		delay(2);
	}

	return 0;
}

File: flex/buzzer/python/beep.py
import RPi.GPIO as GPIO
import time
import signal
import sys

#Piezo buzzer - pin 31 on Raspberry Pi corresponds to BCM GPIO 22
BeepPin = 31

def close(signal, frame):
	GPIO.output(BeepPin, GPIO.LOW)
        sys.exit(0)

# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)
GPIO.setwarnings(False)

signal.signal(signal.SIGINT, close)

# set up GPIO output channel
GPIO.setup(BeepPin, GPIO.OUT)

while True:
	GPIO.output(BeepPin, GPIO.HIGH)
	time.sleep(0.000002)
	GPIO.output(BeepPin, GPIO.LOW)
	time.sleep(0.000002)
GPIO.cleanup() 
